katsdpcal/reduction.py

```diff
@@ -858,5 +915,48 @@ def flush_pipeline(ts, parameters, solution_stores):
     solution_stores : dict of :class:`~.CalSolutionStore`-like
         Solution stores for the capture block, indexed by solution type
     """
-    logger.info('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
-    logger.info('Flushed pipeline')
+    # TODO: num_chunks should enter through paramters
+    #Group the frequency channels into this many sections
+    num_chunks = 16
+    # Extract bandpass gains from solution stores
+    b_solutions = solution_stores['B_POINTING'].get_range(start_time=0, end_time = time.time())
+    # Extract some some commonly used constants from the TS and parameters
+    # Middle time for each dump
+    mid_times = b_solutions.times
+    # TODO The following ideally needs katdal sensor framework for interpolation
+    target_str = ts.get_range('cbf_target', et = mid_times[0])[0][0]
+    target = katpoint.Target(target_str)
+    # Atmospheric conditions
+    soltime = np.mean(mid_times)
+    pres = ts.get_range('anc_air_pressure', et = soltime)[0][0]
+    temp = ts.get_range('anc_air_temperature', et = soltime)[0][0]
+    humi = ts.get_range('anc_air_relative_humidity', et = soltime)[0][0]
+    ants = parameters['antennas']
+    dump_period = ts['int_time']
+    channel_freqs = parameters['channel_freqs']
+    pols = parameters['pol_ordering']
+    
+    # Calculate offset (x,y) co-ordinates for each pointing
+    offsets = get_offsets(ts, parameters, target, mid_times, temp, pres, humi)
+    # Extract gains per pointing offset, per receptor and per frequency chunk.
+    data_points = pointing.get_offset_gains(b_solutions.values(), offsets, ants, channel_freqs,
```

Comment:
It's not a function, just an attribute...
```suggestion
    data_points = pointing.get_offset_gains(b_solutions.values, offsets, ants, channel_freqs,
```